src/checkpointd/runtime.rs

```diff
@@ -0,0 +1,336 @@
+use crate::farcaster_core::consensus::Encodable;
+use farcaster_core::swap::SwapId;
+use lmdb::{Cursor, Transaction as LMDBTransaction};
+use std::path::PathBuf;
+use std::{
+    any::Any,
+    collections::{HashMap, HashSet},
+    convert::{TryFrom, TryInto},
+    io::{self, Write},
+    ptr::swap_nonoverlapping,
+    str::FromStr,
+};
+use strict_encoding::StrictDecode;
+use strict_encoding::StrictEncode;
+
+use crate::swapd::get_swap_id;
+use crate::Endpoints;
+use crate::LogStyle;
+use bitcoin::hashes::{ripemd160, Hash};
+
+use crate::{
+    rpc::{
+        request::{
+            self, BitcoinAddress, Commit, Keys, MoneroAddress, Msg, Params, Reveal, Token, Tx,
+        },
+        Request, ServiceBus,
+    },
+    syncerd::SweepXmrAddress,
+};
+use crate::{CtlServer, Error, Service, ServiceConfig, ServiceId};
+use colored::Colorize;
+use internet2::TypedEnum;
+use microservices::esb::{self, Handler};
+use request::{LaunchSwap, NodeId};
+
+pub fn run(config: ServiceConfig, data_dir: PathBuf) -> Result<(), Error> {
+    let runtime = Runtime {
+        identity: ServiceId::Checkpoint,
+        checkpoints: CheckpointGetSet::new(data_dir).unwrap(),
+        pending_checkpoint_chunks: map![],
+    };
+
+    Service::run(config, runtime, false)
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+struct CheckpointChunk {
```

Comment:
I was reluctant to implement a generic message chunker, because I do not want this to be a permanent solution. The permanent solution is fixing internet2 . That said, I'd be happy to instead of chunking the checkpoint data, chunk the enitre request and rename it accordingly. Then the generic names line more up with the more generic nature of our Request enum. Is this what you are asking for?